### PR TITLE
Updating public API documentation

### PR DESCRIPTION
**Pull request by brawner**
This updates the API documentation of these packages to help bump the documentation category of their Quality Level to 1.

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

File: rosidl_typesupport_fastrtps_c/include/rosidl_typesupport_fastrtps_c/wstring_conversion.hpp [hunk 1]
```diff
@@ -23,10 +23,20 @@
...
+ */
 ROSIDL_TYPESUPPORT_FASTRTPS_C_PUBLIC
 void u16string_to_wstring(
   const rosidl_runtime_c__U16String & u16str, std::wstring & wstr);
 
+/// Convert a std::wstring into a `rosidl_runtime_c__U16String`.
+/**
+ * \param[in] wstr The std::wstring to copy from.
+ * \param[in,out] u16str The u16string to copy to.
+ */
```
Comment (ahcorde) [inline rosidl_typesupport_fastrtps_c/include/rosidl_typesupport_fastrtps_c/wstring_conversion.hpp:39]: \return

File: rosidl_typesupport_fastrtps_cpp/include/rosidl_typesupport_fastrtps_cpp/wstring_conversion.hpp [hunk 2]
```diff
@@ -22,9 +22,19 @@
...
+ * \param[in,out] wstr The std::wstring to copy to.
+ */
 ROSIDL_TYPESUPPORT_FASTRTPS_CPP_PUBLIC
 void u16string_to_wstring(const std::u16string & u16str, std::wstring & wstr);
 
+/// Convert a std::wstring into a std::u16string.
+/**
+ * \param[in] wstr The std::wstring to copy from.
+ * \param[in,out] u16str The std::u16string to copy to.
+ */
```
Comment (ahcorde) [inline rosidl_typesupport_fastrtps_cpp/include/rosidl_typesupport_fastrtps_cpp/wstring_conversion.hpp:37]: \return

File: rosidl_typesupport_fastrtps_cpp/include/rosidl_typesupport_fastrtps_cpp/message_type_support.h [hunk 3]
```diff
@@ -19,25 +19,51 @@
...
 typedef struct message_type_support_callbacks_t
 {
+  /// The C++ namespace of this message.
   const char * message_namespace_;
+
+  /// The typename of this message.
   const char * message_name_;
 
-  // Function for message serialization
+  /// Function for message serialization
```
Comment (Blast545) [inline rosidl_typesupport_fastrtps_cpp/include/rosidl_typesupport_fastrtps_cpp/message_type_support.h:34]: This and below are pointer functions, shouldn't this be reflected in the description? Either replacing with callback or using "Pointer to function"
```suggestion
  /// Callback for message serialization
```

File: rosidl_typesupport_fastrtps_c/include/rosidl_typesupport_fastrtps_c/wstring_conversion.hpp (hunk 1 above)
Comment (brawner) [reply rosidl_typesupport_fastrtps_c/include/rosidl_typesupport_fastrtps_c/wstring_conversion.hpp:40]: Added.

File: rosidl_typesupport_fastrtps_cpp/include/rosidl_typesupport_fastrtps_cpp/message_type_support.h (hunk 3 above)
Comment (brawner) [reply rosidl_typesupport_fastrtps_cpp/include/rosidl_typesupport_fastrtps_cpp/message_type_support.h]: Fixed.

File: rosidl_typesupport_fastrtps_cpp/include/rosidl_typesupport_fastrtps_cpp/wstring_conversion.hpp (hunk 2 above)
Comment (brawner) [reply rosidl_typesupport_fastrtps_cpp/include/rosidl_typesupport_fastrtps_cpp/wstring_conversion.hpp:38]: Fixed.

**Comment by brawner** [comment]
Running for linters:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10764/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6156/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8762/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10652/)

**Comment by brawner** [comment]
I didn't see that build.ros2.org was testing the linting of these files. The failure in the ci.ros2.org tests was fixed by the force push. As can be seen in this passing test.

[Build Status](http://build.ros2.org/job/Fpr__rosidl_typesupport_fastrtps__ubuntu_focal_amd64/9/)

**Pull request merged**